packages/core/src/VNodes/CharNode.ts

```diff
@@ -1,10 +1,67 @@
 import { VNode, VNodeType, FormatType } from './VNode';
 
+/**
+ * This "phantom type" is there to ensure that the type `Char` is only generated
+ * through the use of the function `makeChar`, so as to force going through the
+ * length check.
+ */
+type InternalChar<T> = { valid: true } & string;
+export type Char = InternalChar<{}>;
+/**
+ * Return a Char type from a string of length 1 (validating the type).
+ *
+ * @param char
+ */
+export function makeChar(char: string): Char {
+    if (char.length === 1) {
+        return char as Char;
+    } else {
+        throw new Error('Cannot make a Char out of anything else than a string of length 1.');
+    }
+}
+
```

Comment:
The logic is too complicated to understand for just checking that the char has is only of length one.
I would simply check it in the constructor or in a getter.

Reply:
Also removed already.